keyboards/junco/junco_sync.c

```diff
@@ -0,0 +1,286 @@
+// Copyright 2022 Dane Skalski (@Daneski13)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+/* Only relevant if junco sync is enabled */
+#ifdef JUNCO_SYNC_ENABLE
+#    include "junco_sync.h"
+#    include "print.h"
+#    include "transactions.h"
+#    include "split_util.h"
+#    include "junco.h"
+
+/* Sync common keyboard data */
+
+// Pointer to the user's user_config.raw. If the user wants to sync their own user config,
+// set this to the address of user_config.raw in the keyboard_post_init after reading the
+// user_config from eeprom `user_config_raw_ptr = &user_config.raw`.
+uint32_t* user_config_raw_ptr = NULL;
+
+// Data type for sending data to slave
+typedef struct {
+    uint32_t kb_conf;     // Current keyboard config
+    uint32_t default_lay; // Current default layer, SPLIT_LAYER_STATE_ENABLE is not persistent and may not sync in time
+    uint32_t user_conf;   // If the user has a config...
```

Comment:
Is there a reason for syncing this, as opposed to letting the user handle it?

Reply:
One of the goals of the project is for "plug-and-play" between the sides of the split and the PCB design had that in mind, so the EEPROM settings between the sides of the split should have the same settings. Made sense to implement the common settings at the keyboard level given that goal so the user doesn't have to mess with it within their keymap.